platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,237 @@
+pages for each React application.** The `vets-website` build script
+would then update the landing page with references to the correct
+JavaScript bundle and CSS files to bootstrap the application.
+
+The empty React app landing page would then be deleted or otherwise
+prevented from being served to production.
+
+### Beneficial side effects
+- The `vets-website` build will be faster both locally and in Jenkins
+  - Because the build script wouldn't perform any transformations on
+    the content, this would save about half the build time for
+    developers on their local machines (~20s)
+	- Assuming the cache and the `localhost` buildtype are used
+  - All the assets from Drupal (PDFs, JPEGs, etc.) will be bundled up
+    in the single content tarball, so instead of making 475 individual
+    requests (as of the time of this writing), there will be only a
+    single request
+	- Saving upwards of 40 - 50 seconds
+	- Saving all the unchanged assets again and again into S3 is not
+      ideal
+  - When querying for the latest content, the separated build approach
+    would save upwards of a full minute on `vets-website` builds since
+    it wouldn't have to wait for Drupal to parse the _massive_ GraphQL
+    query, put to gether the result, and send back the multi-megabyte
+    response
+  - The full Jenkins build on `vets-website` would no longer perform
+    the sitemap accessibility tests, which may speed it up
+	- Since these tests are performed in parallel with the normal e2e
+      tests, it's unclear
+	- These tests will still be run—just in the content build because
+      it's a content concern
+- `vets-website` will be isolated from any changes to the way content
+  is built
+  - Example 1: Rather than querying for all the content at once, the
+    build could query for only the _new_ content
+  - Example 2: Whoever manages the content build in the future can
+    decide to skip any pages with broken links, and `vets-website`
+    won't be burdened with the complexity of that decision
+	  
+### How this would affect the CMS team
+I _think_ the impact would be minimal. Rather than using the full
+`vets-website` in their integration environment, they'd use the new
+content building repo.
+
+The CMS team can also trigger a content-only deploy from inside
+Drupal. If this is done before the latest content build is finished,
+the deploy may not have all the latest changes.
+
+## Risks
+- The current integration is _so_ complex; it would be easy to
+  overlook something while implementing the changes
+
+## Questions
+- Who owns the content build repo?
+  - The CMS team?
+  - VSP?
+  - Global UX
+    - Given their relationship with the content, this may make the most sense
+- Who owns managing the content build Jenkins pipeline?
+  - The same team that owns the repo?
+- What triggers a content build?
+  - Could be a daily job
+  - Could be triggered manually from inside Jenkins or even Drupal
```

Comment:
Drupal can already trigger builds on Jenkins for updating content, I'd say that's probably the only primary way we'd need to support.

Reply:
:thinking: Are you saying that when content is updated in Drupal, it can trigger a Jenkins build?

I think if we go that route, that might quickly overwhelm Jenkins as we get _tons_ (hundreds?) of content editors making changes throughout the day.

Reply:
I believe the current process is that you can manually trigger a content build from Drupal. It's not automatic after every change.